browser/typescript/v2/react-app/src/ampli/index.ts

```diff
@@ -0,0 +1,839 @@
+    'String Const WIth Quotes': '"String "Const With" Quotes"',
+    'String Int Const': 0,
+  };
+}
+
+export class EventWithDifferentCasingTypes implements BaseEvent {
+  event_type = 'event withDifferent_CasingTypes';
+
+  constructor(public event_properties: EventWithDifferentCasingTypesProperties) {}
+}
+
+export class EventWithEnumTypes implements BaseEvent {
+  event_type = 'Event With Enum Types';
+
+  constructor(public event_properties: EventWithEnumTypesProperties) {}
+}
+
+export class EventWithOptionalArrayTypes implements BaseEvent {
+  event_type = 'Event With Optional Array Types';
+
+  constructor(public event_properties?: EventWithOptionalArrayTypesProperties) {}
+}
+
+export class EventWithOptionalProperties implements BaseEvent {
+  event_type = 'Event With Optional Properties';
+
+  constructor(public event_properties?: EventWithOptionalPropertiesProperties) {}
+}
+
+export class EventWithTemplateProperties implements BaseEvent {
+  event_type = 'Event With Template Properties';
+
+  constructor(public event_properties: EventWithTemplatePropertiesProperties) {}
+}
+
+// prettier-ignore
+export class Ampli {
+  private disabled: boolean = false;
+  private amplitude?: AmplitudeClient;
+
+  get client(): AmplitudeClient {
+    this.isInitializedAndEnabled();
+    return this.amplitude!;
+  }
+
+  get isLoaded(): boolean {
+    return this.amplitude != null;
+  }
+
+  private isInitializedAndEnabled(): boolean {
+    if (!this.isLoaded) {
+      console.error('ERROR: Ampli is not yet initialized. Have you called ampli.load() on app start?');
+      return false;
+    }
+    return !this.disabled;
+  }
+
+  /**
+   * Initialize the Ampli SDK. Call once when your application starts.
+   * @param options Configuration options to initialize the Ampli SDK with.
+   */
+  load(options?: LoadOptions) {
+    this.disabled = options?.disabled ?? false;
+
+    if (this.isLoaded) {
+      console.warn('WARNING: Ampli is already initialized. Ampli.load() should be called once at application startup.');
+      return;
+    }
+
+    const env = options?.environment ?? 'development';
+    const apiKey = options?.client?.apiKey ?? ApiKey[env];
+
+    if (options?.client?.instance) {
+      this.amplitude = options?.client?.instance;
+    } else if (apiKey) {
+      this.amplitude = amplitude;
+      return this.amplitude?.init(apiKey, undefined, { ...DefaultOptions, ...options?.client?.options });
+    } else {
+      console.error("ERROR: ampli.load() requires 'environment', 'client.apiKey', or 'client.instance'");
+    }
+  }
+
+  /**
+   * Identify a user and set user properties.
+   *
+   * @param userId The user's id.
+   * @param properties The user properties.
+   * @param options Optional event options.
+   */
+  identify(
+    userId: string | undefined,
+    properties: IdentifyProperties,
+    options?: IdentifyOptions,
+  ) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    const event: IdentifyEvent = {
```

Comment:
What's the event here for?

Reply:
This event does not need I think. I will remove it.